### PR TITLE
61103 add default autoload core

### PR DESCRIPTION
**Pull request by pbearne**
Trac ticket: https://core.trac.wordpress.org/ticket/61103

### CODE REVIEW

File: src/wp-admin/includes/upgrade.php [hunk 1]
```diff
@@ -1788,7 +1788,7 @@ function upgrade_330() {
 			$sidebars_widgets                   = $_sidebars_widgets;
 			unset( $_sidebars_widgets );
 
-			// Intentional fall-through to upgrade to the next version.
+		// Intentional fall-through to upgrade to the next version.
```
Comment (mukeshpanchal27) [inline src/wp-admin/includes/upgrade.php]:
```suggestion
			// Intentional fall-through to upgrade to the next version.
```
PHPCS fix

File: src/wp-admin/includes/ajax-actions.php [hunk 2]
```diff
@@ -231,15 +231,15 @@ function wp_ajax_wp_compression_test() {
 			if ( is_multisite() ) {
 				update_site_option( 'can_compress_scripts', 0 );
 			} else {
-				update_option( 'can_compress_scripts', 0, 'yes' );
+				update_option( 'can_compress_scripts', 0, 'on' );
```
Comment (adamsilverstein) [inline src/wp-admin/includes/ajax-actions.php]: update_option accepts 'yes' or 'no' or true/false. Why does this PR use 'on' and 'off'?

Comment (swissspidy) [reply src/wp-admin/includes/ajax-actions.php]: `on` and `off` are new additions in 6.6. See https://core.trac.wordpress.org/ticket/42441 / https://core.trac.wordpress.org/changeset/57920

Comment (joemcgill) [reply src/wp-admin/includes/ajax-actions.php]: That's not exactly correct. `"on"` and `"off"` are the stored DB values, but the function should use boolean `true`/`false` values. For backwards compat, we still support `"yes"` and `"no"`.

**Comment by pbearne** [comment]
I have replaced the on/off with booleans

As I have gone through these I have set any option that looks like it is wp-admin only to not autoload this should help the front-end load a little bit

File: src/wp-admin/includes/upgrade.php [hunk 3]
```diff
@@ -988,7 +991,7 @@ function upgrade_110() {
 
 	$time_difference = $all_options->time_difference;
 
-		$server_time = time() + gmdate( 'Z' );
+	$server_time 	 = time() + gmdate( 'Z' );
 	$weblogger_time  = $server_time + $time_difference * HOUR_IN_SECONDS;
```
Comment (mukeshpanchal27) [inline src/wp-admin/includes/upgrade.php]:
```suggestion
	$server_time     = time() + gmdate( 'Z' );
	$weblogger_time  = $server_time + $time_difference * HOUR_IN_SECONDS;
```

Fix PHPCS

File: src/wp-admin/includes/upgrade.php [hunk 4]
```diff
@@ -2396,7 +2399,45 @@ function upgrade_650() {
...
+			'recently_activated',
+			'https_detection_errors',
+			'fresh_site',
+			'upload_path',
+			'admin_email',
+			'admin_email_lifespan',
+		);
 
+		$autoload = array_fill_keys( $options_ids, 'no' );
+		wp_set_option_autoload_values( $autoload );
```
Comment (mukeshpanchal27) [inline src/wp-admin/includes/upgrade.php]:
```suggestion
		wp_set_options_autoload( $options_ids, 'no' );
```
The function will take care.

File: src/wp-admin/includes/upgrade.php [hunk 5]
```diff
@@ -2396,7 +2399,45 @@ function upgrade_650() {
 		wp_set_option_autoload_values( $autoload );
 	}
 }
+/**
+ * Executes changes made in WordPress 6.5.0.
```
Comment (mukeshpanchal27) [inline src/wp-admin/includes/upgrade.php]:
```suggestion
 * Executes changes made in WordPress 6.7.0.
```

File: src/wp-admin/includes/ajax-actions.php [hunk 6]
```diff
@@ -2487,7 +2487,7 @@ function wp_ajax_delete_inactive_widgets() {
 		$id_base      = implode( '-', $pieces );
 		$widget       = get_option( 'widget_' . $id_base );
 		unset( $widget[ $multi_number ] );
-		update_option( 'widget_' . $id_base, $widget );
+		update_option( 'widget_' . $id_base, $widget, false );
```
Comment (mukeshpanchal27) [inline src/wp-admin/includes/ajax-actions.php]: Nice!

File: src/wp-admin/includes/class-plugin-upgrader.php [hunk 7]
```diff
@@ -262,7 +262,7 @@ public function upgrade( $plugin, $args = array() ) {
 
 		if ( isset( $past_failure_emails[ $plugin ] ) ) {
 			unset( $past_failure_emails[ $plugin ] );
-			update_option( 'auto_plugin_theme_update_emails', $past_failure_emails );
+			update_option( 'auto_plugin_theme_update_emails', $past_failure_emails, false );
```
Comment (mukeshpanchal27) [inline src/wp-admin/includes/class-plugin-upgrader.php]: I don't think we needs to change it, In https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-admin/includes/schema.php#L578C3-L613 core will set `off`.

File: src/wp-includes/option.php [hunk 8]
```diff
@@ -659,6 +659,57 @@ function wp_prime_site_option_caches( array $options ) {
...
 
+
+/**
+ * Primes specific network options for the current network into the cache with a single database query.
+ *
+ * Only network options that do not already exist in cache will be loaded.
+ *
+ * If site is not multisite, then call wp_prime_option_caches().
+ *
+ * @since 6.6.0
```
Comment (mukeshpanchal27) [inline src/wp-includes/option.php]:
```suggestion
 * @since 6.7.0
```

File: src/wp-includes/option.php [hunk 9]
```diff
@@ -659,6 +659,57 @@ function wp_prime_site_option_caches( array $options ) {
...
+		'admin_email_lifespan',
+	);
+	/**
+	 * Filters a list of options to be loaded for WP-admin pages.
+	 *
+	 * @since 6.7.0
+	 *
+	 * @param array $default_core_admin_options Array with admin options to be loaded.
+	 *
+	 * @return array Array with admin options to be loaded.
```
Comment (mukeshpanchal27) [inline src/wp-includes/option.php]:
```suggestion
	 * @param array $default_core_admin_options Array with admin options to be loaded.
	 * @return array Array with admin options to be loaded.
```

**Review by mukeshpanchal27 — COMMENTED** [review]
Thanks @pbearne for the updates. Left some feedbacks.

**Comment by paul** [comment]
Could you fix your bot that keeps mentioning me? I'm getting dozens of notifications a month from this project. https://github.com/WordPress/wordpress-develop/pulls?q=sort%3Aupdated-desc+%22%40paul%22

**Comment by peterwilsoncc** [comment]
> Could you fix your bot that keeps mentioning me? I'm getting dozens of notifications a month from this project. https://github.com/WordPress/wordpress-develop/pulls?q=sort%3Aupdated-desc+%22%40paul%22

@pbearne Are you able to look at the details in your git config so the bot stops pinging the wrong Paul. It's currently set to `paul bearne <paul>` which is causing the bot to be confused.

If you add `pbearne@git.wordpress.org` to your GitHub profile and set that as the email address for WordPress-Develop the commits will be assigned correctly.

**Comment by pbearne** [comment]
> If you add `pbearne@git.wordpress.org` to your GitHub profile and set that as the email address for WordPress-Develop the commits will be assigned correctly.

updated

File: src/wp-admin/includes/class-plugin-upgrader.php (hunk 7 above)
Comment (felixarntz) [inline src/wp-admin/includes/class-plugin-upgrader.php]: The `auto_plugin_theme_update_emails` option is set during WordPress installation (with autoload `false`), so there's no need to specify `false` here. The default will ensure it remains the autoload value it was previously using.

Please change that throughout the rest of the PR as well.

File: src/wp-admin/includes/misc.php [hunk 10]
```diff
@@ -352,7 +352,7 @@ function update_recently_edited( $file ) {
 		$oldfiles[] = $file;
 	}
 
-	update_option( 'recently_edited', $oldfiles );
+	update_option( 'recently_edited', $oldfiles, false );
```
Comment (felixarntz) [inline src/wp-admin/includes/misc.php]: See above: This is already set to `false` during installation, so no need to pass this here.

File: src/wp-admin/includes/plugin.php [hunk 11]
```diff
@@ -711,7 +711,7 @@ function activate_plugin( $plugin, $redirect = '', $network_wide = false, $silen
 			$current   = get_option( 'active_plugins', array() );
 			$current[] = $plugin;
 			sort( $current );
-			update_option( 'active_plugins', $current );
+			update_option( 'active_plugins', $current, false );
```
Comment (felixarntz) [inline src/wp-admin/includes/plugin.php]: This looks problematic to me. The active plugins are fetched on basically every single WordPress request, so this _needs_ to be autoloaded.

However, similar to my previous comments, the value here should be omitted entirely, since this option is already populated during WordPress installation (with autoload `true`), so it'll always be correct.

File: src/wp-admin/includes/plugin.php [hunk 12]
```diff
@@ -1304,7 +1304,7 @@ function uninstall_plugin( $plugin ) {
 	if ( file_exists( WP_PLUGIN_DIR . '/' . dirname( $file ) . '/uninstall.php' ) ) {
 		if ( isset( $uninstallable_plugins[ $file ] ) ) {
 			unset( $uninstallable_plugins[ $file ] );
-			update_option( 'uninstall_plugins', $uninstallable_plugins );
+			update_option( 'uninstall_plugins', $uninstallable_plugins, false );
```
Comment (felixarntz) [inline src/wp-admin/includes/plugin.php]: See above, this is already handled by the WordPress installation, so it should be omitted.

File: src/wp-admin/includes/schema.php [hunk 13]
```diff
@@ -614,7 +614,7 @@ function populate_options( array $options = array() ) {
 
 	// In case it is set, but blank, update "home".
 	if ( ! __get_option( 'home' ) ) {
-		update_option( 'home', $guessurl );
+		update_option( 'home', $guessurl, true );
```
Comment (felixarntz) [inline src/wp-admin/includes/schema.php]: See above, this should be removed as it is configured during WordPress installation.

File: src/wp-admin/includes/upgrade.php [hunk 14]
```diff
@@ -2396,7 +2399,44 @@ function upgrade_650() {
...
+ * @since 6.7.0
+ *
+ * @global int  $wp_current_db_version The old (current) database version.
+ */
+function upgrade_670() {
+	global $wp_current_db_version;
+
+	if ( $wp_current_db_version < 57160 ) {
+		$options_ids = array(
+			'auto_plugin_theme_update_emails',
```
Comment (felixarntz) [inline src/wp-admin/includes/upgrade.php]: This is already set to not autoload during installation, so it should be removed from here.

File: src/wp-admin/includes/upgrade.php [hunk 15]
```diff
@@ -2396,7 +2399,44 @@ function upgrade_650() {
...
+			'recently_activated',
+			'_wp_suggested_policy_text_has_changed',
+			'dashboard_widget_options',
+			'ftp_credentials',
+			'recently_edited',
+			'adminhash',
+			'nav_menu_options',
+			'active_plugins',
```
Comment (felixarntz) [inline src/wp-admin/includes/upgrade.php]: See above, this must be autoloaded, so needs to be removed from here.

File: src/wp-admin/includes/upgrade.php [hunk 16]
```diff
@@ -2396,7 +2399,44 @@ function upgrade_650() {
...
+			'uninstall_plugins',
```
Comment (felixarntz) [inline src/wp-admin/includes/upgrade.php]: See above, this is already not autoloaded during installation, so should be removed.

File: src/wp-admin/includes/upgrade.php [hunk 17]
```diff
@@ -2396,7 +2399,44 @@ function upgrade_650() {
...
+			'admin_email',
+			'recently_activated',
+			'https_detection_errors',
+			'fresh_site',
+			'upload_path',
+			'admin_email',
+			'admin_email_lifespan',
+		);
 
+		wp_set_options_autoload( $options_ids, 'no' );
```
Comment (felixarntz) [inline src/wp-admin/includes/upgrade.php]: Let's use the boolean, as that is encouraged now. Related: https://core.trac.wordpress.org/ticket/61929

File: src/wp-admin/network/site-info.php [hunk 18]
```diff
@@ -98,7 +98,7 @@
 
 	if ( $old_home_host === $existing_details->domain && $old_home_parsed['path'] === $existing_details->path ) {
 		$new_home_url = untrailingslashit( sanitize_url( $blog_data['scheme'] . '://' . $new_details->domain . $new_details->path ) );
-		update_option( 'home', $new_home_url );
+		update_option( 'home', $new_home_url, true );
```
Comment (felixarntz) [inline src/wp-admin/network/site-info.php]: See above, let's remove.

File: src/wp-admin/network/site-info.php [hunk 19]
```diff
@@ -107,7 +107,7 @@
 
 	if ( $old_site_host === $existing_details->domain && $old_site_parsed['path'] === $existing_details->path ) {
 		$new_site_url = untrailingslashit( sanitize_url( $blog_data['scheme'] . '://' . $new_details->domain . $new_details->path ) );
-		update_option( 'siteurl', $new_site_url );
+		update_option( 'siteurl', $new_site_url, true );
```
Comment (felixarntz) [inline src/wp-admin/network/site-info.php]: See above, let's remove.

File: src/wp-admin/network/site-settings.php [hunk 20]
```diff
@@ -46,7 +46,7 @@
 		if ( 0 === $key || is_array( $val ) || in_array( $key, $skip_options, true ) ) {
 			continue; // Avoids "0 is a protected WP option and may not be modified" error when editing blog options.
 		}
-		update_option( $key, $val );
+		update_option( $key, $val, true );
```
Comment (felixarntz) [inline src/wp-admin/network/site-settings.php]: This should remain without a value because we have no idea what the option is and therefore whether or not to autoload it. Alternatively, explicitly pass `null` to indicate the value should be left as is.

File: src/wp-admin/options-privacy.php [hunk 21]
```diff
@@ -52,7 +52,7 @@ static function ( $body_class ) {
 
 	if ( 'set-privacy-page' === $action ) {
 		$privacy_policy_page_id = isset( $_POST['page_for_privacy_policy'] ) ? (int) $_POST['page_for_privacy_policy'] : 0;
-		update_option( 'wp_page_for_privacy_policy', $privacy_policy_page_id );
+		update_option( 'wp_page_for_privacy_policy', $privacy_policy_page_id, true );
```
Comment (felixarntz) [inline src/wp-admin/options-privacy.php]: See above, this is already set during installation and therefore should be removed.

File: src/wp-admin/options-reading.php [hunk 22]
```diff
@@ -74,12 +74,12 @@
 <table class="form-table" role="presentation">
 	<?php
 	if ( 'posts' !== get_option( 'show_on_front' ) ) :
-		update_option( 'show_on_front', 'posts' );
+		update_option( 'show_on_front', 'posts', true );
```
Comment (felixarntz) [inline src/wp-admin/options-reading.php]: See above, this should be removed as it is set during installation.

File: src/wp-admin/options.php [hunk 23]
```diff
@@ -63,7 +63,7 @@
 		&& hash_equals( $new_admin_details['hash'], $_GET['adminhash'] )
 		&& ! empty( $new_admin_details['newemail'] )
 	) {
-		update_option( 'admin_email', $new_admin_details['newemail'] );
+		update_option( 'admin_email', $new_admin_details['newemail'], false );
```
Comment (felixarntz) [inline src/wp-admin/options.php]: Not sure if this is safe to _not_ autoload. It is set during WordPress installation to be autoloaded, so we should definitely not override this here without further consideration.

I think it's best to revert this and stick to the current default passed during installation.

File: src/wp-admin/options.php [hunk 24]
```diff
@@ -341,7 +341,7 @@
 				}
 				$value = wp_unslash( $value );
 			}
-			update_option( $option, $value );
+			update_option( $option, $value, true );
```
Comment (felixarntz) [inline src/wp-admin/options.php]: This should definitely be left out. We don't know what the option is so we don't know whether to autoload it. Either this should be removed or we explicitly pass `null` to leave the value as is, or leave the decision to Core.

File: src/wp-content/themes/twentyfourteen/inc/featured-content.php [hunk 25]
```diff
@@ -268,7 +268,7 @@ public static function delete_post_tag( $tag_id ) {
 
 		$settings['tag-id'] = 0;
 		$settings           = self::validate_settings( $settings );
-		update_option( 'featured-content', $settings );
+		update_option( 'featured-content', $settings, true );
```
Comment (felixarntz) [inline src/wp-content/themes/twentyfourteen/inc/featured-content.php]: This is a default theme change, let's leave those out of the PR as it should only be concerned with update _core's_ usages.

File: src/wp-includes/class-wp-customize-setting.php [hunk 26]
```diff
@@ -656,7 +656,7 @@ protected function set_root_value( $value ) {
 			if ( isset( self::$aggregated_multidimensionals[ $this->type ][ $this->id_data['base'] ]['autoload'] ) ) {
 				$autoload = self::$aggregated_multidimensionals[ $this->type ][ $this->id_data['base'] ]['autoload'];
 			}
-			return update_option( $id_base, $value, $autoload );
+			return update_option( $id_base, $value, $autoload, true );
```
Comment (felixarntz) [inline src/wp-includes/class-wp-customize-setting.php]: This is invalid, the `$autoload` is already the 3rd parameter. Please revert.

File: src/wp-includes/class-wp-xmlrpc-server.php [hunk 27]
```diff
@@ -4330,7 +4330,7 @@ public function wp_setOptions( $args ) {
 				continue;
 			}
 
-			update_option( $this->blog_options[ $o_name ]['option'], wp_unslash( $o_value ) );
+			update_option( $this->blog_options[ $o_name ]['option'], wp_unslash( $o_value ), true );
```
Comment (felixarntz) [inline src/wp-includes/class-wp-xmlrpc-server.php]: See above, we don't know what the options are so we shouldn't make a decision here on whether to autoload them. This should be reverted.

File: src/wp-includes/ms-blogs.php [hunk 28]
```diff
@@ -460,11 +460,11 @@ function update_blog_option( $id, $option, $value, $deprecated = null ) {
 	}
 
 	if ( get_current_blog_id() === $id ) {
-		return update_option( $option, $value );
+		return update_option( $option, $value, true );
```
Comment (felixarntz) [inline src/wp-includes/ms-blogs.php]: See above, please revert.

File: src/wp-includes/ms-blogs.php [hunk 29]
```diff
@@ -460,11 +460,11 @@ function update_blog_option( $id, $option, $value, $deprecated = null ) {
...
 	}
 
 	switch_to_blog( $id );
-	$return = update_option( $option, $value );
+	$return = update_option( $option, $value, true );
```
Comment (felixarntz) [inline src/wp-includes/ms-blogs.php]: See above, please revert.

File: src/wp-includes/ms-deprecated.php [hunk 30]
```diff
@@ -642,17 +642,17 @@ function install_blog( $blog_id, $blog_title = '' ) {
 		}
 	}
 
-	update_option( 'siteurl', $siteurl );
-	update_option( 'home', $home );
+	update_option( 'siteurl', $siteurl, true );
+	update_option( 'home', $home, true );
```
Comment (felixarntz) [inline src/wp-includes/ms-deprecated.php]: See above, please revert.

File: src/wp-includes/option.php [hunk 31]
```diff
@@ -659,6 +659,56 @@ function wp_prime_site_option_caches( array $options ) {
...
+	 * Filters a list of options to be loaded for WP-admin pages.
+	 *
+	 * @since 6.7.0
+	 *
+	 * @param array $default_core_admin_options Array with admin options to be loaded.
+	 * @return array Array with admin options to be loaded.
+	 */
+	$options = apply_filters( 'wp_prime_wp_admin_option_caches', $default_core_admin_options );
+	wp_prime_network_option_caches( null, $options );
+}
```
Comment (felixarntz) [inline src/wp-includes/option.php]: I don't think we should implement a function like this. Almost all of these options are used in one or two specific places, so it's wasteful to load all of them throughout all WP Admin. It would effectively replicate the current problem just scoped to WP Admin.

Options should only be loaded where they're needed if possible. I think the few changes of not autoload certain options in this PR already work well without this, as all modified options (except some, where I left comments) are used only in one admin screen or section.

File: src/wp-includes/plugin.php [hunk 32]
```diff
@@ -936,7 +936,7 @@ function register_uninstall_hook( $file, $callback ) {
 
 	if ( ! isset( $uninstallable_plugins[ $plugin_basename ] ) || $uninstallable_plugins[ $plugin_basename ] !== $callback ) {
 		$uninstallable_plugins[ $plugin_basename ] = $callback;
-		update_option( 'uninstall_plugins', $uninstallable_plugins );
+		update_option( 'uninstall_plugins', $uninstallable_plugins, false );
```
Comment (felixarntz) [inline src/wp-includes/plugin.php]: See above, this is already configured during installation.

File: src/wp-includes/version.php [hunk 33]
```diff
@@ -23,7 +23,7 @@
  *
  * @global int $wp_db_version
  */
-$wp_db_version = 57155;
+$wp_db_version = 57160;
```
Comment (felixarntz) [inline src/wp-includes/version.php]: Why is this change needed?

File: src/wp-login.php [hunk 34]
```diff
@@ -515,7 +515,7 @@ function wp_login_viewport_meta() {
 	$url = dirname( set_url_scheme( 'http://' . $_SERVER['HTTP_HOST'] . $_SERVER['PHP_SELF'] ) );
 
 	if ( get_option( 'siteurl' ) !== $url ) {
-		update_option( 'siteurl', $url );
+		update_option( 'siteurl', $url, true );
```
Comment (felixarntz) [inline src/wp-login.php]: See above, this is not needed.

File: src/wp-login.php [hunk 35]
```diff
@@ -619,7 +619,7 @@ function wp_login_viewport_meta() {
 			}
 
 			if ( $remind_interval > 0 ) {
-				update_option( 'admin_email_lifespan', time() + $remind_interval );
+				update_option( 'admin_email_lifespan', time() + $remind_interval, false );
```
Comment (felixarntz) [inline src/wp-login.php]: Like my comment with `admin_email`, let's not change this here as it's a bit risky to anticipate where it is used. Plus it shouldn't differ from how it is set during WordPress installation.

File: src/wp-admin/includes/ajax-actions.php (hunk 6 above)
Comment (felixarntz) [inline src/wp-admin/includes/ajax-actions.php]: Let's leave this change for https://core.trac.wordpress.org/ticket/42743. It is its own problem to resolve and would make this PR more complicated.

File: src/wp-admin/widgets-form.php [hunk 36]
```diff
@@ -211,7 +211,7 @@
 			$id_base      = implode( '-', $pieces );
 			$widget       = get_option( 'widget_' . $id_base );
 			unset( $widget[ $multi_number ] );
-			update_option( 'widget_' . $id_base, $widget );
+			update_option( 'widget_' . $id_base, $widget, false );
```
Comment (felixarntz) [inline src/wp-admin/widgets-form.php]: See above, let's leave this for https://core.trac.wordpress.org/ticket/42743.

**Review by felixarntz — CHANGES_REQUESTED** [review]
@pbearne I did a thorough review of all the options you're (potentially) modifying here.

My most important feedback is: We should only add `true` or `false` to `update_option()` for options that aren't already added to the database during WordPress installation. Any such options will always already be set when calling `update_option()`.

It is only mandatory to pass an `$autoload` value to `add_option()` because here we're inserting a new option. On `update_option()`, `$autoload` however is only needed if it _may_ pass through to `add_option()` or if we explicitly want to change the autoload value from what it was before. Otherwise it should be `null`.

Also: We shouldn't pass `true` on `update_option()` calls that are for any arbitrary options. For that, the default `null` is intended, which is to leave it up to WordPress core to make the decision.

I think most of the options that are _not_ part of the WordPress installation and that you changed to no longer autoload, it makes sense because they're only used in 1-2 specific admin sections. There are a few though I left comments on too where I think this is too risky and should be separately considered.

File: src/wp-admin/includes/class-plugin-upgrader.php (hunk 7 above)
Comment (pbearne) [reply src/wp-admin/includes/class-plugin-upgrader.php]: done

File: src/wp-admin/includes/misc.php (hunk 10 above)
Comment (pbearne) [reply src/wp-admin/includes/misc.php]: fixed

File: src/wp-admin/includes/plugin.php (hunk 11 above)
Comment (pbearne) [reply src/wp-admin/includes/plugin.php]: fixed

File: src/wp-includes/version.php (hunk 33 above)
Comment (pbearne) [reply src/wp-includes/version.php]: We were updating the DB at one point, reverted

File: src/wp-login.php (hunk 34 above)
Comment (pbearne) [reply src/wp-login.php]: reverted

File: src/wp-admin/includes/plugin.php (hunk 12 above)
Comment (pbearne) [reply src/wp-admin/includes/plugin.php]: fixed

File: src/wp-admin/includes/schema.php (hunk 13 above)
Comment (pbearne) [reply src/wp-admin/includes/schema.php]: fixed

File: src/wp-login.php (hunk 35 above)
Comment (pbearne) [reply src/wp-login.php]: fixed

File: src/wp-admin/widgets-form.php (hunk 36 above)
Comment (pbearne) [reply src/wp-admin/widgets-form.php]: yes

File: src/wp-admin/includes/ajax-actions.php (hunk 6 above)
Comment (pbearne) [reply src/wp-admin/includes/ajax-actions.php]: I agree

File: src/wp-admin/includes/class-plugin-upgrader.php [hunk 37]
```diff
@@ -444,7 +444,7 @@ public function bulk_upgrade( $plugins, $args = array() ) {
 			unset( $past_failure_emails[ $plugin ] );
 		}
 
-		update_option( 'auto_plugin_theme_update_emails', $past_failure_emails );
+		update_option( 'auto_plugin_theme_update_emails', $past_failure_emails, false );
```
Comment (felixarntz) [inline src/wp-admin/includes/class-plugin-upgrader.php]: This still needs to be reverted, see https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731865969.
```suggestion
		update_option( 'auto_plugin_theme_update_emails', $past_failure_emails );
```

File: src/wp-admin/includes/misc.php (hunk 10 above)
Comment (felixarntz) [inline src/wp-admin/includes/misc.php]: This still needs to be reverted, see https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731876165.
```suggestion
	update_option( 'recently_edited', $oldfiles );
```

File: src/wp-admin/includes/plugin.php [hunk 38]
```diff
@@ -842,7 +842,7 @@ function deactivate_plugins( $plugins, $silent = false, $network_wide = null ) {
 	}
 
 	if ( $do_blog ) {
-		update_option( 'active_plugins', $current );
+		update_option( 'active_plugins', $current, false );
```
Comment (felixarntz) [inline src/wp-admin/includes/plugin.php]: Per https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731880434, this needs to be reverted too.
```suggestion
		update_option( 'active_plugins', $current );
```

File: src/wp-admin/includes/plugin.php [hunk 39]
```diff
@@ -1070,7 +1070,7 @@ function validate_active_plugins() {
 	$plugins = get_option( 'active_plugins', array() );
 	// Validate vartype: array.
 	if ( ! is_array( $plugins ) ) {
-		update_option( 'active_plugins', array() );
+		update_option( 'active_plugins', array(), false );
```
Comment (felixarntz) [inline src/wp-admin/includes/plugin.php]: See above:
```suggestion
		update_option( 'active_plugins', array() );
```

File: src/wp-admin/includes/upgrade.php [hunk 40]
```diff
@@ -2397,7 +2400,41 @@ function upgrade_650() {
...
+function upgrade_670() {
+	global $wp_current_db_version;
+
+	if ( $wp_current_db_version < 57160 ) {
+		$options_ids = array(
+			'recently_activated',
+			'_wp_suggested_policy_text_has_changed',
+			'dashboard_widget_options',
+			'ftp_credentials',
+			'recently_edited',
```
Comment (felixarntz) [inline src/wp-admin/includes/upgrade.php]: Per the previous feedback, this option should not be modified.
```suggestion
```

File: src/wp-admin/includes/upgrade.php [hunk 41]
```diff
@@ -2397,7 +2400,41 @@ function upgrade_650() {
...
+			'adminhash',
+			'nav_menu_options',
+			'wp_force_deactivated_plugins',
+			'delete_blog_hash',
+			'allowedthemes',
+			'admin_email',
```
Comment (felixarntz) [inline src/wp-admin/includes/upgrade.php]: See previous feedback, this still needs to be autoloaded.
```suggestion
```

File: src/wp-admin/includes/upgrade.php [hunk 42]
```diff
@@ -2397,7 +2400,41 @@ function upgrade_650() {
...
+			'recently_activated',
```
Comment (felixarntz) [inline src/wp-admin/includes/upgrade.php]: Duplicate, you already have this on top of the list.
```suggestion
```

File: src/wp-admin/includes/upgrade.php [hunk 43]
```diff
@@ -2397,7 +2400,41 @@ function upgrade_650() {
...
+			'https_detection_errors',
+			'fresh_site',
+			'upload_path',
+			'admin_email',
+			'admin_email_lifespan',
```
Comment (felixarntz) [inline src/wp-admin/includes/upgrade.php]: See previous feedback, these should still be autoloaded. On the other hand, the list is missing the `recovery_keys` option.
```suggestion
			'recovery_keys',
```

File: src/wp-admin/options-privacy.php [hunk 44]
```diff
@@ -103,7 +103,7 @@ static function ( $body_class ) {
 				'error'
 			);
 		} else {
-			update_option( 'wp_page_for_privacy_policy', $privacy_policy_page_id );
+			update_option( 'wp_page_for_privacy_policy', $privacy_policy_page_id, true );
```
Comment (felixarntz) [inline src/wp-admin/options-privacy.php]: This still needs to be reverted per https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731897669.
```suggestion
			update_option( 'wp_page_for_privacy_policy', $privacy_policy_page_id );
```

File: src/wp-admin/options-reading.php [hunk 45]
```diff
@@ -79,7 +79,7 @@
 
 else :
 	if ( 'page' === get_option( 'show_on_front' ) && ! get_option( 'page_on_front' ) && ! get_option( 'page_for_posts' ) ) {
-		update_option( 'show_on_front', 'posts' );
+		update_option( 'show_on_front', 'posts', true );
```
Comment (felixarntz) [inline src/wp-admin/options-reading.php]: https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731897851
```suggestion
		update_option( 'show_on_front', 'posts' );
```

File: src/wp-admin/widgets-form.php (hunk 36 above)
Comment (felixarntz) [inline src/wp-admin/widgets-form.php]: This still needs to be addressed per https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731918565.
```suggestion
			update_option( 'widget_' . $id_base, $widget );
```

File: src/wp-includes/class-wp-widget.php [hunk 46]
```diff
@@ -598,7 +598,7 @@ public function _register_one( $number = -1 ) {
 	 */
 	public function save_settings( $settings ) {
 		$settings['_multiwidget'] = 1;
-		update_option( $this->option_name, $settings );
+		update_option( $this->option_name, $settings, wp_use_widgets_block_editor() );
```
Comment (felixarntz) [inline src/wp-includes/class-wp-widget.php]: Per https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731918565, please revert as this should be handled via https://core.trac.wordpress.org/ticket/42743.
```suggestion
		update_option( $this->option_name, $settings );
```

File: src/wp-includes/ms-deprecated.php [hunk 47]
```diff
@@ -646,13 +646,13 @@ function install_blog( $blog_id, $blog_title = '' ) {
 	update_option( 'home', $home );
 
 	if ( get_site_option( 'ms_files_rewriting' ) ) {
-		update_option( 'upload_path', UPLOADBLOGSDIR . "/$blog_id/files" );
+		update_option( 'upload_path', UPLOADBLOGSDIR . "/$blog_id/files", false );
```
Comment (felixarntz) [inline src/wp-includes/ms-deprecated.php]: This should remain autoloaded as it may be regularly accessed in multisite.
```suggestion
		update_option( 'upload_path', UPLOADBLOGSDIR . "/$blog_id/files" );
```

File: src/wp-includes/ms-deprecated.php [hunk 48]
```diff
@@ -646,13 +646,13 @@ function install_blog( $blog_id, $blog_title = '' ) {
...
 	} else {
-		update_option( 'upload_path', get_blog_option( get_network()->site_id, 'upload_path' ) );
+		update_option( 'upload_path', get_blog_option( get_network()->site_id, 'upload_path' ), false );
```
Comment (felixarntz) [inline src/wp-includes/ms-deprecated.php]: See above.
```suggestion
		update_option( 'upload_path', get_blog_option( get_network()->site_id, 'upload_path' ) );
```

File: src/wp-includes/ms-deprecated.php [hunk 49]
```diff
@@ -646,13 +646,13 @@ function install_blog( $blog_id, $blog_title = '' ) {
...
 	}
 
-	update_option( 'blogname', wp_unslash( $blog_title ) );
-	update_option( 'admin_email', '' );
+	update_option( 'blogname', wp_unslash( $blog_title ), true );
+	update_option( 'admin_email', '', false );
```
Comment (felixarntz) [inline src/wp-includes/ms-deprecated.php]: Per previous feedback, these should remain autoloaded as is (and are already set up during installation).
```suggestion
	update_option( 'blogname', wp_unslash( $blog_title ) );
	update_option( 'admin_email', '' );
```

File: src/wp-includes/widgets.php [hunk 50]
```diff
@@ -1099,7 +1099,7 @@ function wp_set_sidebars_widgets( $sidebars_widgets ) {
 		$sidebars_widgets['array_version'] = 3;
 	}
 
-	update_option( 'sidebars_widgets', $sidebars_widgets );
+	update_option( 'sidebars_widgets', $sidebars_widgets, true );
```
Comment (felixarntz) [inline src/wp-includes/widgets.php]: Better to leave this as is, and handle it in https://core.trac.wordpress.org/ticket/42743 if needed.
```suggestion
	update_option( 'sidebars_widgets', $sidebars_widgets );
```

File: src/wp-login.php (hunk 35 above)
Comment (felixarntz) [inline src/wp-login.php]: This still needs to be reverted per https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731915673.
```suggestion
				update_option( 'admin_email_lifespan', time() + $remind_interval );
```

File: src/wp-admin/includes/upgrade.php [hunk 51]
```diff
@@ -2397,7 +2400,41 @@ function upgrade_650() {
...
+			'recently_edited',
+			'adminhash',
+			'nav_menu_options',
+			'wp_force_deactivated_plugins',
+			'delete_blog_hash',
+			'allowedthemes',
+			'admin_email',
+			'recently_activated',
+			'https_detection_errors',
+			'fresh_site',
```
Comment (felixarntz) [inline src/wp-admin/includes/upgrade.php:2426]: There is one more instance where `'fresh_site'` updates needs to be modified, in `wp-admin/includes/upgrade.php`.

**Review by felixarntz — COMMENTED** [review]
@pbearne There are a few more instances from my previous review that still need to be reverted, but it's getting close.

I can make some of these updates inline.

File: src/wp-admin/includes/upgrade.php (hunk 51 above)
Comment (felixarntz) [reply src/wp-admin/includes/upgrade.php:2426]: Updated in https://github.com/WordPress/wordpress-develop/pull/6465/commits/28e9a8779a768d17544157b8d39bd71305e2d950

**Review by felixarntz — APPROVED** [review]

**Comment by felixarntz** [comment]
Committed in https://core.trac.wordpress.org/changeset/58975
